storage/copy_impl.go

```diff
@@ -64,18 +64,11 @@ func (c copyImpl) CopyRaw(ctx context.Context, source, destination DataReference
 	return nil
 }
 
-func newCopyMetrics(scope promutils.Scope) copyMetrics {
-	return copyMetrics{
+func newCopyMetrics(scope promutils.Scope) *copyMetrics {
+	return &copyMetrics{
 		CopyLatency:                  labeled.NewStopWatch("overall", "Overall copy latency", time.Millisecond, scope, labeled.EmitUnlabeledMetric),
 		ComputeLengthLatency:         labeled.NewStopWatch("length", "Latency involved in computing length of content before writing.", time.Millisecond, scope, labeled.EmitUnlabeledMetric),
 		WriteFailureUnrelatedToCache: scope.MustNewCounter("write_failure_unrelated_to_cache", "Raw store write failures that are not caused by ErrFailedToWriteCache"),
 		ReadFailureUnrelatedToCache:  scope.MustNewCounter("read_failure_unrelated_to_cache", "Raw store read failures that are not caused by ErrFailedToWriteCache"),
 	}
 }
-
-func newCopyImpl(store RawStore, metricsScope promutils.Scope) copyImpl {
```

Comment:
can we keep this constructor but pass in the set of already initialized copyMetrics?